src/redux/categories/categoriesSlice.js

```diff
@@ -0,0 +1,14 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState: [],
+  reducers: {
+    checkStatus: () => {
+      const initialState = 'Under construction';
+      return initialState;
+    },
```

Comment:
- [ ] Ouch! this is a close one but as part of the project requirements you should check the status from the action payload before returning the message in the reducer, This means the reducer should accept `state` and `action` as parameters as shown below 👇.
```jsx
reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },

```

Reply:
- Refactored to check the status from the action payload before returning the message in the reducer.